travharv/executor.py

```diff
@@ -46,6 +50,10 @@ def assert_all_paths(self):
                subjects given for each task per config"""
         )
         for task in self.tasks:
+            task_execution_report = TaskExecutionReport(
+                self.execution_report.rdf_store_access
+            )
+            self.execution_report.report_to_store(task_execution_report)
```

Comment:
coin almost starting to drop on these 'report_to_store' methods

starts to feel like most of these objects have some kind of a `parent`or `context` they belong to hierarchically
why not just pass that in the __init__ of each?

Reply:
... was still chewing on this one, went all over the place, am now coming back to a variant that makes things just plain and easy


drop these abstract but not reused `report_to_store` methods and simply got for `add_task_report`

and other, better descriptive names for the others contexts

also feels like there is no reporting just logging in these methods, and those feel like they could be moved to the constructors of the sublevels instead?  that would lower the entanglement between all these

```python
            task_execution_report = TaskExecutionReport()
            self.execution_report.add_task_report(task_execution_report)
```

and yes, I still don´t get why all these levels need access to the store